File: backend/dotnet-webapi/src/WorldCup.Api/Dtos/MatchWithCityDto.cs
using WorldCup.Api.Models;

namespace WorldCup.Api.Dtos;

public class MatchWithCityDto
{
    public string Id { get; set; } = string.Empty;
    public Team HomeTeam { get; set; } = null!;
    public Team AwayTeam { get; set; } = null!;
    public City City { get; set; } = null!;
    public string Kickoff { get; set; } = string.Empty;
    public string Group { get; set; } = string.Empty;
    public int MatchDay { get; set; }
    public double TicketPrice { get; set; }

    public static MatchWithCityDto FromEntity(Match match) => new()
    {
        Id = match.Id,
        HomeTeam = match.HomeTeam,
        AwayTeam = match.AwayTeam,
        City = match.City,
        Kickoff = match.Kickoff,
        Group = match.GroupName,
        MatchDay = match.MatchDay,
        TicketPrice = match.TicketPrice
    };
}

File: backend/java-spring/src/main/java/com/unosquare/worldcup/dto/MatchWithCityDTO.java
package com.unosquare.worldcup.dto;

import com.unosquare.worldcup.model.City;
import com.unosquare.worldcup.model.Match;
import com.unosquare.worldcup.model.Team;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class MatchWithCityDTO {
    private String id;
    private Team homeTeam;
    private Team awayTeam;
    private City city;
    private LocalDateTime kickoff;
    private String group;
    private int matchDay;
    private Double ticketPrice;

    public static MatchWithCityDTO fromEntity(Match match) {
        return new MatchWithCityDTO(
                match.getId(),
                match.getHomeTeam(),
                match.getAwayTeam(),
                match.getCity(),
                match.getKickoff(),
                match.getGroupName(),
                match.getMatchDay(),
                match.getTicketPrice()
        );
    }
}
